pkg/apis/autoscaling/v1alpha1/metric_validation_test.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+	"github.com/google/go-cmp/cmp"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"knative.dev/pkg/apis"
+)
+
+func TestMetricValidation(t *testing.T) {
+	tests := []struct {
+		name string
+		m    *Metric
+		want *apis.FieldError
```

Comment:
This is still `*FieldError`, but up to you.

Reply:
`*FieldError` is present in `knative.dev/pkg/apis` so 
it should be `*apis.FieldError` only `*FieldError` wont work

Reply:
I meant use `string` rather than the error itself, but doesn't matter much.